python/pyarrow/array.pyx

```diff
@@ -360,3 +425,35 @@ cdef class Table:
             data.append(<object> arr)
 
         return pd.DataFrame(dict(zip(names, data)), columns=names)
+
+    def name(self):
+        self._check_nullptr()
+        return frombytes(self.table.name())
+
+    def schema(self):
+        raise NotImplementedError()
+
+    def column(self, index_or_name):
+        self._check_nullptr()
+        index = index_or_name
+        # TODO(uwe): Implement retrevial by name
+        cdef Column column = Column()
+        column.init(self.table.column(index))
+        return column
+
+    def columns(self):
+        for i in range(self.num_columns()):
+            yield self.column(i)
+
+    def num_columns(self):
+        self._check_nullptr()
+        return self.table.num_columns()
+
+    def num_rows(self):
```

Comment:
Perhaps make these properties also -- can you add `__len__` that calls `num_rows`?